packages/near-membrane-base/src/environment.ts

```diff
@@ -1,192 +1,181 @@
-                    };
-                }
+            const configurable = ObjectHasOwnProperty.call(blueDescriptor, 'configurable')
+                ? !!blueDescriptor.configurable
+                : undefinedSymbol;
+            const enumerable = ObjectHasOwnProperty.call(blueDescriptor, 'enumerable')
+                ? !!blueDescriptor.enumerable
+                : undefinedSymbol;
+            const writable = ObjectHasOwnProperty.call(blueDescriptor, 'writable')
+                ? !!blueDescriptor.writable
+                : undefinedSymbol;
+            const valuePointer = ObjectHasOwnProperty.call(blueDescriptor, 'value')
+                ? this.blueGetTransferableValue(blueDescriptor.value)
+                : undefinedSymbol;
+            const getPointer = ObjectHasOwnProperty.call(blueDescriptor, 'get')
+                ? this.blueGetTransferableValue(blueDescriptor.get)
+                : undefinedSymbol;
+            const setPointer = ObjectHasOwnProperty.call(blueDescriptor, 'set')
+                ? this.blueGetTransferableValue(blueDescriptor.set)
+                : undefinedSymbol;
+            // installing descriptor into the red side
+            this.redCallableDefineProperty(
+                oPointer,
+                key,
+                configurable,
+                enumerable,
+                writable,
+                valuePointer,
+                getPointer,
+                setPointer
+            );
+        }
+    }
 
-                if (typeof blueDescriptor.set === 'function') {
-                    redDescriptor.set = function set(v: RedValue): void {
-                        // if a global setter is invoke, the value will be use as it
-                        // is as the result of the getter operation
-                        currentBlueGetter = () => v;
-                    };
+    lazyRemap(o: ProxyTarget, keys: PropertyKey[]) {
+        const enumerablePropertyKeys = Object.keys(o); // except symbols
+        const keysTuple: (PropertyKey | boolean)[] = [];
+        for (let i = 0, len = keys.length; i < len; i += 1) {
+            const key = keys[i];
+            let isEnumerable = true;
+            if (typeof key === 'symbol') {
+                if (ObjectPropertyIsEnumerable.call(o, key)) {
+                    isEnumerable = true;
                 }
             }
-            // "as any" supresses the "Type 'symbol' cannot be used as an index type."
-            redDescriptors[key as any] = redDescriptor;
+            // Skip index keys for magical descriptors of frames on the window proxy.
+            // TODO: this applies to all objects rather than just `window`, it might
+            // be a problem.
+            else if (RegExpTest(frameGlobalNamesRegExp, key as string)) {
+                // eslint-disable-next-line no-continue
+                continue;
+            } else {
+                isEnumerable = enumerablePropertyKeys.includes(key as string);
```

Comment:
👆 We should use the `ArrayIncludes` generic helper instead of prototype `.includes`